lib/sasl/src/systools_relup.erl

```diff
@@ -454,12 +454,12 @@ create_add_app_scripts(FromRel, ToRel, RU0s, W0s) ->
 %%
 %% XXX ToApps not used.
 %%
-create_remove_app_scripts(FromRel, ToRel, RU0s, W0s) -> 
+append_remove_app_scripts(FromRel, ToRel, RU0s, W0s) -> 
     RemovedNs = [N || {N, _V, _T} <- FromRel#release.applications,
 		      not lists:keymember(N, 1, ToRel#release.applications)],
     %% io:format("Removed apps: ~p~n", [RemovedNs]),
     RUs = [[{remove_application, N}] || N <- RemovedNs],
-    {RUs ++ RU0s, W0s}.
+    { RU0s ++ RUs, W0s}.
```

Comment:
this is the clincher here right?

Reply:
and the change to `del_module` instead of `remove` in `systools_rc`.